src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -61,6 +63,61 @@ internal sealed class State
                 SingleReader = true,
                 SingleWriter = true,
             });
+
+            public void RemoveStream(MsQuicStream stream)
+            {
+                bool closing;
+                lock (this)
+                {
+                    StreamCount--;
+                    closing = _closingGCHandle != IntPtr.Zero && StreamCount == 0;
+                }
+
+                if (closing)
+                {
+                    Handle?.Dispose();
+                    try
+                    {
+                        GCHandle gcHandle = GCHandle.FromIntPtr(_closingGCHandle);
```

Comment:
Why we use IntPtr here instead of GCHandle itself?

Reply:
GCHandle is struct so I could not make it nullable.

Reply:
ok. I can do that and use `IsAllocated` to check if the handle is active.

Reply:
updated. Since the `_state` needs to live at least as long as the connection it self, I moved the GCHandle there. It will either be freed when connection is disposed or when last stream is gone. No more need for copy or lookup.